addons/stock/report/report_stock_forecasted.py

```diff
@@ -39,8 +39,8 @@ def _move_draft_domain(self, product_template_ids, product_variant_ids, wh_locat
 
     def _move_confirmed_domain(self, product_template_ids, product_variant_ids, wh_location_ids):
         in_domain, out_domain = self._move_domain(product_template_ids, product_variant_ids, wh_location_ids)
-        out_domain += [('state', 'not in', ['draft', 'cancel', 'done'])]
-        in_domain += [('state', 'not in', ['draft', 'cancel', 'done'])]
+        out_domain += [('state', 'in', ['waiting', 'assigned', 'confirmed', 'partially_available'])]
+        in_domain += [('state', 'in', ['waiting', 'assigned', 'confirmed', 'partially_available'])]
```

Comment:
Nice, 
About your commit message, you should also say that the 'not in' operator on a selection doesn't allow PostgreSQL to use the index on this selection (if it exists). (Also, the fix is in stock, not mrp). In this case, now, PostgreSQL might use the index on `state` whatever it improves its plan.